### PR TITLE
[RW-4878][risk=no] Convert criteria selection list to React

### PR DESCRIPTION
**Pull request by dolbeew**
- Created new React component `SelectionList`, previously part of `ModalComponent`, since the list will eventually be moved into the sidebar.
- Converted  `SelectionInfo` component to React.
Screen Shot 2020-05-06 at 9 06 05 AM

@als364  to review React conversion
@freemabd  to deploy and test CB functionality

---
**PR checklist**

- [x] This PR meets the Acceptance Criteria in the JIRA story
- [x] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have run and tested this change locally

### CODE REVIEW

File: ui/src/app/cohort-search/modal/modal.component.html [hunk 1]
```diff
@@ -129,43 +128,16 @@
...
-                  [disabled]="disableFlag"
-                  (click)="finish()"
-                  class="btn btn-primary">
-                  Finish
-                </button>
-              </div>
-            </div>
+            <crit-selection-list
+              [back]="back"
+              [cancel]="close"
```
Comment (als364) [inline ui/src/app/cohort-search/modal/modal.component.html:133]: I'd prefer for this to be named consistently - if the only thing this function does is close the list then name it `close` throughout.

File: ui/src/app/cohort-search/modal/modal.component.html [hunk 2]
```diff
@@ -129,43 +128,16 @@
...
+              [disableFinish]="modifiersDisabled"
```
Comment (als364) [inline ui/src/app/cohort-search/modal/modal.component.html:134]: Same with this - why is it named differently in different files?

File: ui/src/app/cohort-search/selection-list/selection-list.component.tsx [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+      <div style={styles.selectionContainer}>
+        {selections.map((selection, s) =>
+          <SelectionInfo key={s}
+            index={s}
+            selection={selection}
+            removeSelection={() => removeSelection(selection)}/>
+        )}
+      </div>
+      <div style={styles.buttonContainer}>
+        <button style={{...styles.button, background: 'transparent', color: colors.dark, fontSize: '14px'}}
```
Comment (als364) [inline ui/src/app/cohort-search/selection-list/selection-list.component.tsx:187]: Consider using the `Button` component in `buttons.tsx`

File: ui/src/app/cohort-search/selection-list/selection-list.component.tsx [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+          Cancel
+        </button>
+        {this.showNext && <button onClick={() => setView('modifiers')}
+          style={{
+            ...styles.button,
+            ...(selections.length === 0 ? styles.disabled : {}),
+            background: colors.primary, color: colors.white}}>
+          Next
+        </button>}
+        {this.showBack && <button style={{...styles.button, background: colors.primary, color: colors.white}}
```
Comment (als364) [inline ui/src/app/cohort-search/selection-list/selection-list.component.tsx:198]: You get most of these styles for free with the `Button` component.

File: ui/src/app/cohort-search/modal/modal.component.html (hunk 2 above)
Comment (dolbeew) [reply ui/src/app/cohort-search/modal/modal.component.html]: Changed to `modifiersDisabled`.  I renamed it when I separated the list into the new component and didn't rename in the parent (parent is next to be converted to React, so would've been cleaned up then).

File: ui/src/app/cohort-search/selection-list/selection-list.component.tsx (hunk 3 above)
Comment (dolbeew) [reply ui/src/app/cohort-search/selection-list/selection-list.component.tsx]: Good idea, switch all 4 buttons here to use `Button`

File: ui/src/app/cohort-search/selection-list/selection-list.component.tsx (hunk 4 above)
Comment (dolbeew) [reply ui/src/app/cohort-search/selection-list/selection-list.component.tsx]: Switched.

**Pull request merged**